crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -134,16 +134,18 @@ where
     #[inline]
     fn initialize(&mut self, world: &mut World) {
         self.world_id = Some(world.id());
-        self.system_meta.last_change_tick = world.change_tick().wrapping_sub(MAX_CHANGE_AGE);
+        self.system_meta
+            .last_run
+            .set(world.change_tick().tick.wrapping_sub(MAX_CHANGE_AGE));
```

Comment:
I would prefer to see this logic refactored into a method on `Tick` now that we can. It's used in a couple of places, and is rather tricky.

Reply:
Ideally we could make the internal `tick` field completely private.

Reply:
Good idea